lib/FlyBy.pm

```diff
@@ -255,7 +255,11 @@ sub all_keys {
 
 sub values_for_key {
     my ($self, $key) = @_;
-    return (sort { $a cmp $b } keys %{$self->index_sets->{$key}});
+
+    my $set = $self->index_sets->{$key};
+    my @values = grep { my $v = $set->{$_}; $v && %{$v->{elements}} } keys %$set;
```

Comment:
It took a bit but I think I found the real culprit. Lines 100++ read like this:

```perl
    if (not $add_missing_key and not exists $index_sets->{$key}) {
        $result = ($negated) ? $self->_full_set : $self->_full_set->empty_clone;    # Avoiding auto-viv on request
    } else {
```

The `$add_missing_key` variable is a flag that indicates if the key should be added to the index or not. But why the hell is this combined with `and not exists ...`?

So, it should read something like this:

```perl
    if (not $add_missing_key) {
        $result = ($negated) ? $self->_full_set : $self->_full_set->empty_clone;    # Avoiding auto-viv on request
    } elsif (not exists $index_sets->{$key}) {
    	# do something if $add_missing_key is set but the key does not exist in the index.
    } else {
```